### PR TITLE
Unifying domain indicators

### PR DESCRIPTION
**Pull request by reutshal**
## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22111
fixes: https://github.com/demisto/etc/issues/22201#event-3100276076

## Description
- Created a new formatting script which extracts both FQDNs and Domains
- Created a new test playbook for the script
- Changed old domain reputations files names to 5.4.9 rather than 5.5
- The new script is very similar to ExtractFQDNFromUrlAndEmail, you can look at the diff between them.
- Probably going to need to force merge:
1. Changed the id of the reputation-domain file to the new domain
2. Added toVersion 5.4.9 to the old domain in reputations.json file, but apparently demisto-sdk does not recognize the scheme of the file and therefore file validation is failing.

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [X] 5.5.0

## Does it break backward compatibility?
   - [X] No (changing id of a file which is from version 5.5)
 
## Must have
- [ ] Tests
- [ ] Documentation 
- [X] Code Review

### CODE REVIEW

File: Misc/reputations.json [hunk 1]
```diff
@@ -183,7 +183,8 @@
         "DomainReputation"
       ],
       "formatScript": "ExtractDomainFromUrlAndEmail",
-      "fromVersion": "3.6"
+      "fromVersion": "3.6",
+      "toVersion": "5.4.9"
```
Comment (hirshbergshachar) [inline Misc/reputations.json]: From and to version doesn't work in this file so this expiration won't work

File: Misc/reputations_CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ## [Unreleased]
-The **CVE CVSS Score** indicator type is only available in Demisto version 4.5.x and earlier.
+- The **CVE CVSS Score** indicator type is only available in Demisto version 4.5.x and earlier.
+- Added toVersion 5.4.9 to the old domain indicator
```
Comment (hirshbergshachar) [inline Misc/reputations_CHANGELOG.md]: Do we need this in the release notes?

Comment (ronykoz) [reply Misc/reputations_CHANGELOG.md]: No, please remove

File: Misc/reputations.json (hunk 1 above)
Comment (ronykoz) [reply Misc/reputations.json]: shouldn't we also add the new indicator here?

File: Scripts/ExtractDomainAndFQDNFromUrlAndEmail/ExtractDomainAndFQDNFromUrlAndEmail.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
+import demistomock as demisto
```
Comment (ronykoz) [inline Scripts/ExtractDomainAndFQDNFromUrlAndEmail/ExtractDomainAndFQDNFromUrlAndEmail.py]: this script is basically the the same as the old one for extract FQDN + some minor adjustments, right?

File: Misc/reputations.json (hunk 1 above)
Comment (reutshal) [reply Misc/reputations.json]: @ronykoz we don't want to support this file from 5.5 but to create a different file for every indicator.
@hirshbergshachar really? but we want only one domain indicator from 5.5.
So what can we do about the one that's in this file?

File: Misc/reputations_CHANGELOG.md (hunk 2 above)
Comment (reutshal) [reply Misc/reputations_CHANGELOG.md]: Will do

File: Scripts/ExtractDomainAndFQDNFromUrlAndEmail/ExtractDomainAndFQDNFromUrlAndEmail.py (hunk 3 above)
Comment (reutshal) [reply Scripts/ExtractDomainAndFQDNFromUrlAndEmail/ExtractDomainAndFQDNFromUrlAndEmail.py]: Exactly

File: Misc/reputations.json (hunk 1 above)
Comment (hirshbergshachar) [reply Misc/reputations.json]: You should remove it from this file, and add it as a separate file with those to and from versions. You can see the other indicator types for which we did the same

Comment (reutshal) [reply Misc/reputations.json]: @hirshbergshachar So we actually do support expiration from the file.
Added toVersion in the meantime, I will open a PR to delete this file and work only with a separate file to each indicator.

**Comment by reutshal** [comment]
FYI - Added the default mapping for domain from this PR https://github.com/demisto/content/pull/5852

File: Scripts/ExtractDomainAndFQDNFromUrlAndEmail/ExtractDomainAndFQDNFromUrlAndEmail.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  version: -1
+enabled: true
+name: ExtractDomainAndFQDNFromUrlAndEmail
+script: '-'
+system: false
+tags:
+- indicator-format
+timeout: '0'
+type: python
+dockerimage: demisto/tld:latest
```
Comment (glicht) [inline Scripts/ExtractDomainAndFQDNFromUrlAndEmail/ExtractDomainAndFQDNFromUrlAndEmail.yml]: Let's upgrade this image. I will create a new image in dockerfiles.

**Comment by glicht** [comment]
@reutshal build doesn't pass validate files. I tried merging from master but doesn't help.

**Comment by glicht** [comment]
@reutshal I've modified the script to use the new docker and moved everything to py3.

Please add unit tests that check extract from email

**Comment by reutshal** [comment]
> Please make sure to test uploading of the content zip to a clean 5.5 system and also a 5.0 system that we didn't break bc

Tested in a 5.5, will validate in 5.0 ASAP

**Comment by meirwah** [comment]
@reutshal can you share performance results here for this branch ?

**Comment by reutshal** [comment]
@meirwah performance results:
`=== Total Incidents: 50 || Ingestion: 885.544627ms || Complete: 3m5.813846686s ===`
While on the master branch:
`=== Total Incidents: 50 || Ingestion: 787.027519ms || Complete: 2m46.581792233s ===`

**Comment by reutshal** [comment]
@glicht Can you please review?

**Comment by reutshal** [comment]
@kirbles19 Can you please go over the files:
- https://github.com/demisto/content/blob/Unifying_domain_indicators/Layouts/layout-indicatorsDetails-domain_CHANGELOG.md
- https://github.com/demisto/content/blob/Unifying_domain_indicators/Scripts/ExtractDomainAndFQDNFromUrlAndEmail/CHANGELOG.md
- https://github.com/demisto/content/blob/Unifying_domain_indicators/Scripts/ExtractDomainAndFQDNFromUrlAndEmail/ExtractDomainAndFQDNFromUrlAndEmail.yml
- https://github.com/demisto/content/blob/Unifying_domain_indicators/Scripts/ExtractDomainFromUrlFormat/CHANGELOG.md
- https://github.com/demisto/content/blob/Unifying_domain_indicators/Scripts/ExtractFQDNFromUrlAndEmail/CHANGELOG.md
- https://github.com/demisto/content/blob/Unifying_domain_indicators/Scripts/script-UnEscapeURLs_CHANGELOG.md
Thanks!

**Comment by kirbles19** [comment]
@reutshal Amazing work, very few (minor) updates.

**Comment by ronykoz** [comment]
Force merging due to editing old script with no docker images.

**Pull request merged**